packetbeat/protos/dhcpv4/dhcpv4.go

```diff
@@ -86,10 +88,38 @@ func (p *dhcpv4Plugin) parseDHCPv4(pkt *protos.Packet) *beat.Event {
 	v4, err := dhcpv4.FromBytes(pkt.Payload)
 	if err != nil {
 		metricParseFailures.Inc()
-		p.log.Warnw("dropping packet: failed parsing DHCP data", "error", err)
+		p.log.Warnw("Dropping packet: failed parsing DHCP data", "error", err)
 		return nil
 	}
 
+	evt, pbf := pb.NewBeatEvent(pkt.Ts)
+
+	// source/destination (note: this protocol does not produce a bi-flow.)
+	src, dst := common.MakeEndpointPair(pkt.Tuple.BaseTuple, nil)
+	pbf.SetSource(&src)
+	pbf.SetDestination(&dst)
+	pbf.Source.Bytes = int64(len(pkt.Payload))
+
+	if v4.Opcode() == dhcpv4.OpcodeBootReply {
+		// Reverse
+		client, server := ecs.Client(*pbf.Destination), ecs.Server(*pbf.Source)
+		pbf.Client = &client
+		pbf.Server = &server
```

Comment:
Thanks for filling out src/dst, even if the cli/srv pair makes more sense at a conceptual level. 👍